helm-charts/nginx-ingress/templates/post-upgrade-job.yaml

```diff
@@ -18,6 +18,10 @@ metadata:
     app.kubernetes.io/managed-by: {{ .Release.Service }}
     app.kubernetes.io/name: {{ .Release.Name }}
     helm.sh/chart: {{ .Chart.Name }}
+  annotations:
+    "helm.sh/hook": post-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-failed
```

Comment:
should retain the failed job in the failed case. Please remove `hook-failed` in the delete policy.